docs/victorialogs/security-and-lb.md

```diff
@@ -0,0 +1,423 @@
+        drop_src_path_prefix_parts: 1
+```
+
+`extra_filters` and `extra_stream_filters` must be encoded using [percent encoding](https://en.wikipedia.org/wiki/Percent-encoding).
+For example, the query `_stream:{service=frontend-logs}` should be written as `_stream%3A%7Bservice%3Dfrontend-logs%7D`.
+
+Prefer using `extra_stream_filters` over `extra_filters` whenever possible.
+This can improve search query performance because VictoriaLogs
+processes searches using stream filters faster than regular filters.
+
+## Write Authorization
+
+For log writing, [VictoriaLogs single-node](https://docs.victoriametrics.com/victorialogs/#high-availability-ha-setup-with-victorialogs-single-node-instances)
+and [vlinsert](https://docs.victoriametrics.com/victorialogs/cluster/) use the same write API and use HTTP endpoints
+starting with `/insert/`, so when configuring write requests, it is important to set this path for request authorization.
+
+Example vmauth configuration that allows insert requests
+with Basic auth authentication and distributes load between `vlinsert`:
+
+```yaml
+users:
+  - username: foo
+    password: bar
+    url_map:
+      - src_paths: ["/insert/.*"]
+        url_prefix:
+        - "http://vlinsert-1:9428"
+        - "http://vlinsert-2:9428"
+        - "http://vlinsert-3:9428"
+```
+
+Note that vmauth does not allow writing data to multiple clusters, instead, it handles load balancing.
+To replicate data to multiple clusters, use [vlagent](https://docs.victoriametrics.com/victorialogs/vlagent/).
+
+If you trust the writing side, for example when collecting logs within your own cluster, you can set
+the [query parameters](https://docs.victoriametrics.com/victorialogs/data-ingestion/#http-query-string-parameters)
+and [tenant headers](https://docs.victoriametrics.com/victorialogs/#multitenancy) directly at the write level without an extra proxy.
+In such cases, it's usually sufficient to restrict network access so only trusted agents can write to vlinsert or VictoriaLogs single-node.
+
+On the other hand, if you do not trust the writing side, for example, if the logs come from frontend or mobile apps,
+it is very important to secure the write API:
+- Set up a secure HTTPS connection for [vmauth](https://docs.victoriametrics.com/victoriametrics/vmauth/#tls-termination-proxy) and [VictoriaLogs](https://docs.victoriametrics.com/victorialogs/cluster/#security).
+- Place a load balancer before vmauth that can detect bots and DDoS attacks.
+- Consider the [max_concurrent_requests](https://docs.victoriametrics.com/victoriametrics/vmauth/#concurrency-limiting) parameter to control the number of concurrent write requests.
+- Add [monitoring and alerting for vmauth](https://docs.victoriametrics.com/victoriametrics/vmauth/#monitoring) to control the load.
+- Write logs from untrusted applications to separate clusters so that the unpredictable write load does not affect other clusters.
```

Comment:
```suggestion
- Write logs from untrusted applications to dedicated VictoriaLogs instances or clusters so that the unpredictable write load does not affect other instances.
```